### ISSUE TITLE
RStudio Server installation can't install packages

### ISSUE THREAD

**Issue by revodavid** [Azure/aztk#355]
It's not possible to install packages from RStudio Server (after following the directions [here](https://github.com/Azure/aztk/wiki/SparklyR-on-Azure-with-AZTK) and logging in at localhost:8787). For example

```
install.packages("sparklyr")
```

fails with various "Permission Denied" errors. The problem is that R_HOME is set to /usr/local/lib/R/site-library, which is not writeable by the RStudio user. Running `chmod -R a+w /usr/local/lib/R/site-library` as root works around the problem, but it's a hack.

**Comment by jafreck** [comment]
Thanks for pointing this out! We will look into it right away.

**Comment by revodavid** [comment]
The correct solution is for `.libPaths()` to include a local folder (it should be something like "/home/david/R/x86_64-pc-linux-gnu-library/3.4") but I guess the R installation in the image is misconfigured or something.

**Comment by revodavid** [comment]
Hmm, this may in fact be an RStudio bug. Installing literally any other package (try: `install.packages("fortunes")` works as expected, and R creates ~/R/site-library for the rstudio user. But RStudio seems to intercept the sparklyr install, which fails with the permission errors.

**Comment by revodavid** [comment]
I think I figured out what the problem is. If you run `env` "in the terminal you will see:

```
R_LIBS_USER=/usr/local/lib/R/site-library
```

That's where R will attempt to install packages for the `rstudio` user. I guess it does have permissions to write there, but attempting to reinstall an *existing* package, like sparklyr, fails.

**Comment by jafreck** [comment]
That is correct -- the issue only arises for reinstalling packages as permissions in the container were not properly configured. We put out a fix for [this](https://github.com/Azure/aztk/commit/5ceec673467d11288c099d6dee0f680b62bd1e73). Please note that the tags for R Docker images on docker have changed (the wiki will be updated momentarily).

Instead of `aztk/r-base:spark2.2.0` please use `aztk/r-base:spark2.2.0-r3.4.1-base` or just `aztk/r-base:latest`.

Could you confirm that the issue has been fixed?

**Comment by revodavid** [comment]
I got the following error when trying to create with aztk/r-base:latest

```
azure.common.AzureHttpError: The specifed resource name contains invalid characters.
```

And aztk/r-base:spark2.2.0-r3.4.1-base is still giving similar errors for a package already installed:
```
> install.packages("assertthat")
Error in install.packages : Updating loaded packages
> install.packages("assertthat")
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
trying URL 'https://mran.microsoft.com/snapshot/2018-01-24/src/contrib/assertthat_0.2.0.tar.gz'
Content type 'application/octet-stream' length 11612 bytes (11 KB)
==================================================
downloaded 11 KB

* installing *source* package ‘assertthat’ ...
** package ‘assertthat’ successfully unpacked and MD5 sums checked
mv: cannot move '/usr/local/lib/R/site-library/assertthat' to '/usr/local/lib/R/site-library/00LOCK-assertthat/assertthat': Permission denied
Warning in file.copy(f, instdir, TRUE) :
  problem copying ./NAMESPACE to /usr/local/lib/R/site-library/assertthat/NAMESPACE: Permission denied
Warning in file(file, if (append) "a" else "w") :
  cannot open file '/usr/local/lib/R/site-library/assertthat/DESCRIPTION': Permission denied
Error in file(file, if (append) "a" else "w") : 
  cannot open the connection
ERROR: installing package DESCRIPTION failed for package ‘assertthat’
* removing ‘/usr/local/lib/R/site-library/assertthat’
Warning in install.packages :
  installation of package ‘assertthat’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/RtmpG6NSBf/downloaded_packages’
```

**Comment by jafreck** [comment]
The first error is usually due to the name provided for the cluster. It must be at least 3 characters (the error is not clear and should be updated).

The images are still building, however, so the issue will persist until that is finished. I will keep you updated.

**Comment by jafreck** [comment]
@revodavid The images have finished building so this issue should be resolved. Let me know if you are experiencing any problems.

**Comment by jafreck** [comment]
Closing since this is fixed by #341
